src/pretix/api/views/checkin.py

```diff
@@ -283,13 +283,23 @@ def redeem(self, *args, **kwargs):
             else:
                 op = queryset.get(secret=self.kwargs['pk'])
         except OrderPosition.DoesNotExist:
-            self.request.event.log_action('pretix.event.checkin.unknown', data={
+            revoked_matches = list(self.request.event.revoked_secrets.filter(secret=self.kwargs['pk']))
+            if len(revoked_matches) == 0 or not force:
+                self.request.event.log_action('pretix.event.checkin.unknown', data={
+                    'datetime': dt,
+                    'type': type,
+                    'list': self.checkinlist.pk,
+                    'barcode': self.kwargs['pk']
+                }, user=self.request.user, auth=self.request.auth)
+                raise Http404()
+
+            op = revoked_matches[0].position
+            op.order.log_action('pretix.event.checkin.revoked', data={
                 'datetime': dt,
                 'type': type,
                 'list': self.checkinlist.pk,
                 'barcode': self.kwargs['pk']
             }, user=self.request.user, auth=self.request.auth)
```

Comment:
- [ ] `pretix.event.checkin.revoked` logaction does not exist yet
- [ ] Perhaps I'm not in the right mindspace - but should an online-scan of a revoked ticket be declined with "unknown" or actually with "revoked"? (I can see the both as valid usecases)
- [ ] Displaying the whole ticket-secret from the logentry on the dashboard was already a risky business before (due to space constraints) - but with super-long base64-encoded strings that won't wrap, it's almost guaranteed to break the layout